src/ctx.ts

```diff
@@ -0,0 +1,20 @@
+import * as vscode from 'vscode';
+import { SPMPackage } from './package';
+
+export class Ctx {
+	private constructor(
+        public workspaceRoot: string,
```

Comment:
`public` is the default for TypeScript. Is there a reason you've added it here?

Reply:
changing to `public` gives the following error
```
'public' modifier cannot appear on a module or namespace element.
````

Reply:
Ah yes. I think this shorthand syntax (declare properties via the constructor) requires every property to have an explicit access control modifier.